plugin/types.go

```diff
@@ -183,24 +191,33 @@ type Streamable interface {
 	Stream(context.Context) (io.ReadCloser, error)
 }
 
-// BlockReadable is an entry with content that can be read in blocks
+// BlockReadable is an entry with data that can be read in blocks.
+// A BlockReadable entry must set its Size attribute. If you don't set it, the
+// file size will be reported as 0 and reads will return an empty file.
 type BlockReadable interface {
 	Entry
 	Read(ctx context.Context, size int64, offset int64) ([]byte, error)
 }
 
-// Readable is an entry with content that can be read
+// Readable is an entry with data that can be read.
 type Readable interface {
 	Entry
 	Read(context.Context) ([]byte, error)
 }
 
-// Writable is an entry that can write data directly to the entry. It mirrors
-// the WriterAt interface with an added context for handling the lifecycle of
-// remote write operations.
+// Writable is an entry that we can write new data to. What that means can be
+// implementation-specific; it could be overwriting a file, submitting a
+// configuration change to an API, or writing data to a queue. It doesn't
+// support a concept of a partial write.
```

Comment:
Should we add a reminder to plugin authors to include write-semantics for non-file-like entries in the description? Also holds for the external plugin docs.

Reply:
Done.